apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.RequestTemplate.cs

```diff
@@ -0,0 +1,226 @@
+﻿// Copyright 2020 Google LLC
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     https://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Google.Cloud.Storage.V1
+{
+    public sealed partial class UrlSigner
+    {
+        /// <summary>
+        /// Represents the request template to be signed.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// When either of the headers collections are specified, there are certain headers which will be included in the
+        /// signed URL's signature, and therefore must be included in requests made with the created URL. These are the
+        /// Content-MD5 and Content-Type content headers as well as any content or request header with a name starting
+        /// with "x-goog-".
+        /// </para>
+        /// <para>
+        /// If the headers collections are null or empty, no headers are included in the signed URL's signature, so any
+        /// requests made with the created URL must not contain Content-MD5, Content-Type, or any header starting with "x-goog-".
+        /// </para>
+        /// <para>
+        /// Note that when GET is specified as the <see cref="HttpMethod"/> (or it is null, in which case GET is
+        /// used), both GET and HEAD requests can be made with the created signed URL.
+        /// </para>
+        /// </remarks>
+        public sealed class RequestTemplate
+        {
+            /// <summary>
+            /// The name of the bucket. Won't be null.
```

Comment:
Nit: I'd change "Won't be null" to "This will never be null" just as a slightly less abrupt form of words. Just a personal preference though. (Ditto later.)